modules/phase_field/src/kernels/CoupledSusceptibilityTimeDerivativeInterface.C

```diff
@@ -0,0 +1,74 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "CoupledSusceptibilityTimeDerivativeInterface.h"
+
+registerMooseObject("PhaseFieldApp", CoupledSusceptibilityTimeDerivativeInterface);
+
+template <>
+InputParameters
+validParams<CoupledSusceptibilityTimeDerivativeInterface>()
+{
+  InputParameters params = validParams<CoupledSusceptibilityTimeDerivative>();
+  params.addClassDescription(
+      "A modified coupled time derivative Kernel that multiplies the time "
+      "derivative of a coupled variable by a generalized susceptibility and interface normal");
+  return params;
+}
+
+CoupledSusceptibilityTimeDerivativeInterface::CoupledSusceptibilityTimeDerivativeInterface(
+    const InputParameters & parameters)
+  : CoupledSusceptibilityTimeDerivative(parameters), _grad_v(coupledGradient("v"))
+{
+}
+
+Real
+CoupledSusceptibilityTimeDerivativeInterface::computeQpResidual()
+{
+  if (_grad_v[_qp].norm() > 1.0e-6)
+    return _F[_qp] * _v_dot[_qp] * _grad_v[_qp] * _grad_test[_i][_qp] / _grad_v[_qp].norm();
+  else
+    return 0.0;
+}
+
+Real
+CoupledSusceptibilityTimeDerivativeInterface::computeQpJacobian()
+{
+  if (_grad_v[_qp].norm() > 1.0e-6)
+    return _dFdu[_qp] * _v_dot[_qp] * _grad_v[_qp] * _grad_test[_i][_qp] * _phi[_j][_qp] /
+           _grad_v[_qp].norm();
+  else
+    return 0.0;
+}
+
+Real
+CoupledSusceptibilityTimeDerivativeInterface::computeQpOffDiagJacobian(unsigned int jvar)
+{
+  // get the coupled variable jvar is referring to
+  const unsigned int cvar = mapJvarToCvar(jvar);
+
+  if (_grad_v[_qp].norm() > 1.0e-6)
```

Comment:
Compute `_grad_v[_qp].norm() ` once and store it in a variable. Better yet, use `norm_sq()` and compare that to `libMesh::TOLERANCE`. Only if the norm square is larger than the tolerance use `std::sqrt` to compute the norm!